### DISCUSSION TITLE
Skywalking Java Agent Compactability with OAP(8.9.1)

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#9597] [Q&A]
Hi,

Skywalking OAP :8.9.1
Java Agent : 8.12.0 

I wish to know the compatibility of Java agent with old version of OAP. I am using old version of OAP(8.9.1) but i wanted to upgrade the agent to the latest version. Is there any dependency in that. The given version of OAP will be able to collect data from latest agent? Else is there any option to make it compactable that i can add. Or what change need to do for that in OAP. 
Thanks in Advance.

**Comment by wu-sheng** [comment]
There isn't such kind of compatibility guarantee for the case. Because you never knew whether this would work or not. I can't tell the things have not happened.

**Reply by wu-sheng** [reply]
Being reality, and being engineering. Don't expect an impossible thing.

**Reply by AjinThankachan** [reply]
Thanks I understand. But ideally which agent version supports which OAP something similar will be documented is it. If you can give me an idea/document  which version of agent will support which OAP collector so that i can move further. And check on the same. If the agent don't have complete dependency of OAP version i think we can do this without any trouble. 
Thanks once again

**Reply by wu-sheng** [reply]
I don't know. Maybe it works, considering the data-collect-protocol is stable. Check this repo, https://github.com/apache/skywalking-data-collect-protocol. Once it is stable between the versions of agents you choose, you should be safe.

But the whole way to guarantee is, that you tested it. There is no upstream test about this part, which is why I said this is impossible to tell.

**Reply by liyuntian** [reply]
skywalking agent version8.3.0，skywalking server version9.2.0. I can get trace and metrics correctly，

Maybe I haven't encountered any mistakes yet

**Comment by sruby** [comment]
I also plan to upgrade agent to the latest version to support more plugins, what is the result of running after the upgrade? @AjinThankachan

**Reply by Superskyyy** [reply]
The documentation has been there for a long time. https://skywalking.apache.org/docs/main/next/en/setup/service-agent/agent-compatibility/

**Reply by wu-sheng** [reply]
This is not what they are asking. They asked about new agent, but old OAP. This is no guarantee for that.

**Reply by Superskyyy** [reply]
My bad, didn't look carefully.
